lib/LANraragi/Model/Tankoubon.pm

```diff
@@ -0,0 +1,270 @@
+
+use experimental "try";
+
+use strict;
+use warnings;
+use utf8;
+
+use Redis;
+use Mojo::JSON qw(decode_json encode_json);
+use List::Util qw(min);
+
+use LANraragi::Utils::Database qw(redis_encode redis_decode invalidate_cache get_archive_json_multi);
+use LANraragi::Utils::Logging qw(get_logger);
+
+# get_tankoubon_list(page)
+#   Returns a list of all the Tankoubon objects.
+# NEEDS PAGINATION
+sub get_tankoubon_list($page=0) {
+
+    my $redis = LANraragi::Model::Config->get_redis;
+    my $logger = get_logger( "Tankoubon", "lanraragi" );
+
+    # Tankoubons are represented by RG_[timestamp] in DB. Can't wait for 2038!
+    my @tanks = $redis->keys('TANK_??????????');  
+
+    # Jam tanks into an array of hashes
+    my @result;
+    foreach my $key (sort @tanks) {
+        my %data = get_tankoubon($key);
+        push( @result, \%data );
+    }
+
+    # # Only get the first X keys
+    my $keysperpage = LANraragi::Model::Config->get_pagesize;
+
+    # Return total keys and the filtered ones
+    my $total = $#tanks+1;
+    my $start = $page*$keysperpage;
+    my $end = min( $start + $keysperpage - 1, $#result );
+    return ( $total, $#result + 1, @result[ $start .. $end ] );
+
+    #return @result;
+}
+
+# create_tankoubon(name, existing_id)
+#   Create a Tankoubon.
+#   If an existing Tankoubon ID is supplied, said Tankoubon will be updated with the given parameters.
+#   Returns the ID of the created/updated Tankoubon.
+sub create_tankoubon( $name, $tank_id ) {
+
+    my $redis = LANraragi::Model::Config->get_redis;
+
+    # Set all fields of the group object
+    unless ( length($tank_id) ) {
+        $tank_id = "TANK_" . time();
+
+        my $isnewkey = 0;
+        until ($isnewkey) {
+
+            # Check if the group ID exists, move timestamp further if it does
+            if ( $redis->exists($tank_id) ) {
+                $tank_id = "TANK_" . ( time() + 1 );
+            } else {
+                $isnewkey = 1;
+            }
+        }
+    }
+
+    # Default values for new group
+    # Score 0 will be reserved for the name of the tank
+    $redis->zadd( $tank_id, 0, redis_encode($name)); 
+
+    $redis->quit;
+
+    return $tank_id;
+}
+
+# get_tankoubon(tankoubonid, decoded, page)
+#   Returns the Tankoubon matching the given id.
+#   Returns undef if the id doesn't exist.
+sub get_tankoubon($tank_id,$decoded=0,$page=0) {
+
+    my $logger = get_logger( "Tankoubon", "lanraragi" );
+    my $redis  = LANraragi::Model::Config->get_redis;
+    my $keysperpage = LANraragi::Model::Config->get_pagesize;
+
+    if ( $tank_id eq "" ) {
+        $logger->debug("No Tankoubon ID provided.");
+        return ();
+    }
+
+    unless ( length($tank_id) == 15 && $redis->exists($tank_id) ) {
+        $logger->warn("$tank_id doesn't exist in the database!");
+        return ();
+    }
+
+    # Declare some needed variables
+    my %tank;
+    my @archives; 
+    my @limit = split(' ', "LIMIT " . ($keysperpage * $page) . " $keysperpage");
```

Comment:
If this endpoint has pagination, it should return some information as to how many IDs were returned and how many total pages/IDs exist.  

Similarly to the other endpoint, I think it should also have a way to disable pagination and just get all the IDs -- which wouldn't be too expensive unless you have a gigantic tankoubon and add `decoded` to the call.  
(at which point I'd say it's your fault as a client, what the fug are you doing)